src/lcec_el7041_1000.c

```diff
@@ -0,0 +1,713 @@
+
+  hal_data->last_dcm_enable   = 0;
+
+  return 0;
+}
+
+void lcec_el7041_1000_read(struct lcec_slave *s, long period) {
+  lcec_master_t *m = s->master;
+  lcec_el7041_1000_data_t *hal_data = (lcec_el7041_1000_data_t *) s->hal_data;
+  uint8_t *pd = m->process_data;
+  lcec_el7041_1000_chan_t *c;
+  int16_t raw_count, raw_latch, raw_delta;
+
+  // wait for slave to be operational
+  if (!s->state.operational) {
+    hal_data->last_operational = 0;
+    return;
+  }
+
+  // check inputs
+  c = &hal_data->chans[0];
+
+  // check for change in scale value
+  if (*(c->pos_scale) != c->enc_old_scale) {
+      // scale value has changed, test and update it
+      if ((*(c->pos_scale) < 1e-20) && (*(c->pos_scale) > -1e-20)) {
+              // value too small, divide by zero is a bad thing
+              *(c->pos_scale) = 1.0;
+      }
+      // save new scale to detect future changes
+      c->enc_old_scale = *(c->pos_scale);
+      // we actually want the reciprocal
+      c->enc_scale_recip = 1.0 / *(c->pos_scale);
+  }
+
+  // get bit states
+  *(c->ina) = EC_READ_BIT(&pd[c->ina_pdo_os], c->ina_pdo_bp);
+  *(c->inb) = EC_READ_BIT(&pd[c->inb_pdo_os], c->inb_pdo_bp);
+  *(c->inc) = EC_READ_BIT(&pd[c->inc_pdo_os], c->inc_pdo_bp);
+  *(c->inext) = EC_READ_BIT(&pd[c->inext_pdo_os], c->inext_pdo_bp);
+  *(c->sync_err) = EC_READ_BIT(&pd[c->sync_err_pdo_os], c->sync_err_pdo_bp);
+  *(c->expol_stall) = EC_READ_BIT(&pd[c->expol_stall_pdo_os], c->expol_stall_pdo_bp);
+  *(c->tx_toggle) = EC_READ_BIT(&pd[c->tx_toggle_pdo_os], c->tx_toggle_pdo_bp);
+  *(c->count_overflow) = EC_READ_BIT(&pd[c->count_overflow_pdo_os], c->count_overflow_pdo_bp);
+  *(c->count_underflow) = EC_READ_BIT(&pd[c->count_underflow_pdo_os], c->count_underflow_pdo_bp);
+  *(c->latch_c_valid) = EC_READ_BIT(&pd[c->latch_c_valid_pdo_os], c->latch_c_valid_pdo_bp);
+  *(c->latch_ext_valid) = EC_READ_BIT(&pd[c->latch_ext_valid_pdo_os], c->latch_ext_valid_pdo_bp);
+
+  *(c->dcm_ready_to_enable) = EC_READ_BIT(&pd[c->dcm_ready_to_enable_pdo_os], c->dcm_ready_to_enable_pdo_bp);
+  *(c->dcm_ready) = EC_READ_BIT(&pd[c->dcm_ready_pdo_os], c->dcm_ready_pdo_bp);
+  *(c->dcm_warning) = EC_READ_BIT(&pd[c->dcm_warning_pdo_os], c->dcm_warning_pdo_bp);
+  *(c->dcm_error) = EC_READ_BIT(&pd[c->dcm_error_pdo_os], c->dcm_error_pdo_bp);
+  *(c->dcm_move_pos) = EC_READ_BIT(&pd[c->dcm_move_pos_pdo_os], c->dcm_move_pos_pdo_bp);
+  *(c->dcm_move_neg) = EC_READ_BIT(&pd[c->dcm_move_neg_pdo_os], c->dcm_move_neg_pdo_bp);
+  *(c->dcm_torque_reduced) = EC_READ_BIT(&pd[c->dcm_torque_reduced_pdo_os], c->dcm_torque_reduced_pdo_bp);
+  *(c->dcm_din1) = EC_READ_BIT(&pd[c->dcm_din1_pdo_os], c->dcm_din1_pdo_bp);
+  *(c->dcm_din2) = EC_READ_BIT(&pd[c->dcm_din2_pdo_os], c->dcm_din2_pdo_bp);
+  *(c->dcm_sync_err) = EC_READ_BIT(&pd[c->dcm_sync_err_pdo_os], c->dcm_sync_err_pdo_bp);
+  *(c->dcm_tx_toggle) = EC_READ_BIT(&pd[c->dcm_tx_toggle_pdo_os], c->dcm_tx_toggle_pdo_bp);
+
+  hal_data->internal_fault = *(c->dcm_error);
```

Comment:
Mixed usage of multichannel/singlechannel variables. Please use channel seperation in consistent manner (for affected variables) or switch to single channel style.